### PR TITLE
feat(authentication): add `updateProfile` method

### LINKED ISSUE #28: feat(authentication): add method `updateProfile`
**Is your feature request related to a problem? Please describe:**

Currently there is no option to update the user's profile.

**Describe the solution you'd like:**

Add method `updateProfile`.

### PR DESCRIPTION
**Pull request by trancee**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] The changes have been tested successfully.
- [x] A changeset has been created (`npm run changeset`).

Close #28

### CODE REVIEW

**Comment by robingenz** [comment]
LGTM

**Comment by trancee** [comment]
@robingenz I am trying to fix the path in the `README.md` file where you give an example how to use `capacitor.config.ts`:
```ts
/// <reference types="@capacitor/firebase-authentication" />
```
This seems wrong as it should be 
```ts
/// <reference types="@capacitor-firebase/authentication" />
```
But fixing this directly in the `README.md` will revert it back to the wrong one once I called `npm verify:web`. I have not found where it adds this line, maybe you could fix it?

**Comment by robingenz** [comment]
@trancee This is an open issue which i already reported: https://github.com/ionic-team/capacitor-docgen/issues/40
You can just leave it this way for now. I hope it gets fixed soon.

**Comment by trancee** [comment]
@robingenz I was wondering, when the `updateProfile` method successfully updated the profile, there is no `authStateChanged` event emitted and it is not clear what the current state of the profile is. Do you know if it is meant that way, or do we have to call something like `getProfile` to receive the latest update of the profile?

**Comment by robingenz** [comment]
> @robingenz I was wondering, when the `updateProfile` method successfully updated the profile, there is no `authStateChanged` event emitted and it is not clear what the current state of the profile is. Do you know if it is meant that way, or do we have to call something like `getProfile` to receive the latest update of the profile?

You should call `getCurrentUser(...)` to retrieve the users profile (see [Firebase docs](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_profile)).

File: .changeset/warm-eyes-float.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@capacitor-firebase/authentication": minor
+---
+
+feat: add method updateProfile
```
Comment (robingenz) [inline .changeset/warm-eyes-float.md:5]:
```suggestion
feat: add method `updateProfile`
```

File: packages/authentication/src/definitions.ts [hunk 2]
```diff
@@ -374,6 +374,12 @@ export interface FirebaseAuthenticationPlugin {
    * @since 0.1.0
    */
   updatePassword(options: UpdatePasswordOptions): Promise<void>;
+  /**
+   * Updates a user's profile data.
+   *
+   * @since 1.4.0
```
Comment (robingenz) [inline packages/authentication/src/definitions.ts:380]: If you want, we can add this feature to the upcoming release `v1.3.0`.

File: packages/authentication/src/definitions.ts [hunk 3]
```diff
@@ -591,6 +597,24 @@ export interface UpdatePasswordOptions {
...
+   */
+  displayName?: string | null;
+  /**
+   * The user's photo URL.
+   *
+   * @since 1.4.0
+   */
+  photoURL?: string | null;
```
Comment (robingenz) [inline packages/authentication/src/definitions.ts:615]:
```suggestion
  photoUrl?: string | null;
```

I know it's called `photoURL` in the Firebase docs, but I already have it in use as `photoUrl` (see [User](https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/authentication#user)).

**Review by robingenz — CHANGES_REQUESTED** [review]

File: packages/authentication/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/authentication/FirebaseAuthenticationPlugin.java [hunk 4]
```diff
@@ -570,6 +570,22 @@ public void updatePassword(PluginCall call) {
         }
     }
 
+    @PluginMethod
+    public void updateProfile(PluginCall call) {
+        try {
+            String displayName = call.getString("displayName");
+            String photoURL = call.getString("photoURL");
```
Comment (robingenz) [inline packages/authentication/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/authentication/FirebaseAuthenticationPlugin.java:577]:
```suggestion
            String photoURL = call.getString("photoUrl");
```

File: packages/authentication/ios/Plugin/FirebaseAuthenticationPlugin.swift [hunk 5]
```diff
@@ -407,6 +407,24 @@ public class FirebaseAuthenticationPlugin: CAPPlugin {
         })
     }
 
+    @objc func updateProfile(_ call: CAPPluginCall) {
+        let displayName = call.getString("displayName")
+        let photoURL = call.getString("photoURL")
```
Comment (robingenz) [inline packages/authentication/ios/Plugin/FirebaseAuthenticationPlugin.swift:412]:
```suggestion
        let photoURL = call.getString("photoUrl")
```

File: packages/authentication/package.json [hunk 6]
```diff
@@ -48,7 +48,7 @@
     "eslint": "eslint . --ext ts",
     "prettier": "prettier \"**/*.{css,html,ts,js,java}\"",
     "swiftlint": "node-swiftlint",
-    "docgen": "docgen --api FirebaseAuthenticationPlugin --output-readme README.md --output-json dist/docs.json",
+    "docgen": "docgen --api FirebaseAuthenticationPlugin --package @capacitor-firebase --output-readme README.md --output-json dist/docs.json",
```
Comment (robingenz) [inline packages/authentication/package.json:51]: Does this param already work?

File: packages/authentication/src/definitions.ts [hunk 7]
```diff
@@ -591,6 +597,24 @@ export interface UpdatePasswordOptions {
   newPassword: string;
 }
 
+/**
+ * @since 1.4.0
```
Comment (robingenz) [inline packages/authentication/src/definitions.ts:601]:
```suggestion
 * @since 1.3.0
```

**Review by robingenz — CHANGES_REQUESTED** [review]

File: packages/authentication/package.json (hunk 6 above)
Comment (trancee) [reply packages/authentication/package.json:51]: No, it hasn't been merged yet. I actually forgot to remove it again...

**Review by robingenz — APPROVED** [review]
@trancee Thank you for your contribution again!

**Pull request merged**